build.rs

```diff
@@ -472,6 +475,19 @@ fn build_c_code(
     let core_srcs = sources_for_arch(&target.arch)
         .into_iter()
         .filter(|p| !is_perlasm(p))
+        .filter(|p| {
+            if let Some(extension) = p.extension() {
+                // We don't (and can't) use any .S on Windows since MSVC and NASM can't assemble
+                // them.
+                if extension == "S"
+                    && (target.arch == X86_64 || target.arch == X86)
```

Comment:
Can rustc build it perhaps? I don't currently have a win machine, but what happen if you include it as a glob asm?

Reply:
If clang is available then we could use it. We could also preassemble `.S` using clang for Windows, like we do for `.asm` for nasm source files.

BoringSSL doesn't use this (nor nistz256) on Windows so IDK if there are ABI issues. I think there may be some concerns that it doesn't work with unwinding but I'm not sure that's a big deal for us. More investigation is warranted. We do enable nistz256 on Windows in *ring* so it wouldn't be unprecedented.